src/Nix/Effects/Basic.hs

```diff
@@ -308,7 +294,7 @@ defaultDerivationStrict = fromValue @(AttrSet (NValue t f m)) >=> \s -> do
     coerceNixList :: NValue t f m -> m (NValue t f m)
     coerceNixList v = do
       xs <- fromValue @[NValue t f m] v
-      ys <- traverse (\x -> demand x coerceNix) xs
```

Comment:
I've never liked this suggestion of hlint. I rather see `flip demand coerceNix`.

Reply:
It throws me off that you commenting the `master` code, while I think you referring to my patch line.
 
I really like
```
`demand` coerceNix
```

It is in code and mentally - one function operation "less". I have good position/block perception so it is easier mentally to feed into the block / drag the block around in the mind, then switch inputs (& that evokes thinking of what types in that inputs).

I also directly signals that function is used *here* as an infix binary operation. That is easy to process for me.